### DISCUSSION TITLE
Argument of type '{ tags: string[]; }' is not assignable to parameter of type 'TestConfigOverrides'.

### DISCUSSION THREAD

**Discussion by HariPrasad0102** [cypress-io/cypress#24779] [Questions and Help]
Hi, 

I get the following error on testfile.ts when I try to implement the GrepTags:

```
Argument of type '{ tags: string[]; }' is not assignable to parameter of type 'TestConfigOverrides'.
  Object literal may only specify known properties, and 'tags' does not exist in type 'TestConfigOverrides'.ts(2345)
```

![Screenshot 2022-11-22 at 3 21 52 PM](https://user-images.githubusercontent.com/97685046/203391881-96234b81-63b6-4bd6-b0ae-0b9c892a8595.png)

Could someone help me with this ?

**Comment by portyu9** [comment]
Hi @HariPrasad0102 

This was a **typing problem in the `@cypress/grep` package around that Cypress 11 timeframe**, not invalid tag syntax.

The typings were subsequently fixed. On a current Cypress setup, tags on both tests and `describe()` blocks are supported:

```js
describe('some suite', { tags: ['@regression', '@smoke'] }, () => {
  it('does something', { tags: '@smoke' }, () => {
    // ...
  })
})
```

So the preferred fix today is to upgrade to a compatible `@cypress/grep` rather than adding your own TypeScript declaration workaround.

For Cypress 16, use `@cypress/grep` **6.0.0 or newer**. The current Cypress plugin documentation lists `@cypress/grep` 7.x.

Also make sure the package's types are visible to your Cypress TypeScript project if you explicitly restrict `compilerOptions.types`:

```json
{
  "compilerOptions": {
    "types": ["cypress", "@cypress/grep", "node"]
  }
}
```

For an old Cypress 11 project that absolutely cannot be upgraded, declaration-merging `TestConfigOverrides`/`SuiteConfigOverrides` was a temporary workaround, but I would not carry that workaround into a current installation.

Please mark this answer as accepted if it helped, thank you.
